state/scale.go

```diff
@@ -0,0 +1,60 @@
+package state
+
+import (
+	"github.com/hashicorp/nomad-autoscaler/state/status"
+	"github.com/hashicorp/nomad/api"
+)
+
+// statusUpdate is responsible for taking a Nomad JobScaleStatusResponse and
+// updating the autoscaler internal state.
+func (h *Handler) statusUpdate(update *api.JobScaleStatusResponse) {
+
+	// If the job is stopped, remove the scaling policies, remove the scale
+	// status handler and delete the scale status state. Even if the job is
+	// started again, the Autoscaler will gather all the information it
+	// needs so we can clear everything out.
+	switch update.JobStopped {
+	case true:
+		h.PolicyState.DeleteJobPolicies(update.JobID)
+		h.removeStatusHandler(update.JobID)
+		h.statusState.DeleteJob(update.JobID)
+		h.log.Debug("job is stopped, removed internal state", "job_id", update.JobID)
+	default:
+		// Store the new scale status information and move along.
+		h.statusState.SetJob(update)
+		h.log.Debug("set scale status in state", "job_id", update.JobID)
+	}
```

Comment:
does this need to be in a `for {}` loop?

Reply:
It does not. The function is called each time we have an update to process.